R/system.R

```diff
@@ -2334,6 +2334,31 @@ download_data_file <- function(url, type){
   }
 }
 
+#'API that imports multiple same structure Excel files and merge it to a single data frame
+#'@export
+read_excel_files <- function(files, sheet = 1, col_names = TRUE, col_types = NULL, na = "", skip = 0, trim_ws = TRUE, n_max = Inf, use_readxl = NULL, detectDates = FALSE, skipEmptyRows = FALSE, skipEmptyCols = FALSE, check.names = FALSE, tzone = NULL, ...) {
+    # set name to the files so that it can be used for the "id" column created by purrr::map_dfr.
+    files <- setNames(as.list(files), files)
+    df <- purrr::map_dfr(files, exploratory::read_excel_file, sheet = sheet,
+           col_names = col_names,
+           col_types = col_types,
+           na = na,
+           skip = skip,
+           trim_ws = trim_ws,
+           n_max = n_max,
+           use_readxl = use_readxl,
+           detectDates = detecDates,
+           skipEmptyRows = skipEmptyRows,
+           skipEmptyCols = skipEmptyCols,
+           check.names = check.names,
+           tzone = tzone, .id = "exp.file.id") %>% mutate(exp.file.id = basename(exp.file.id)) # extract file name from full path with basename.
+    id_col <- avoid_conflict(colnames(df), "id")
+    # copy internal exp.file.id to the id column.
+    df[[id_col]] <- df[["exp.file.id"]]
+    # drop internal column and move the id column to the very beginning.
+    df %>% dplyr::select(!!!rlang::syms(id_col), dplyr::everything(), -exp.file.id)
```

Comment:
!!rlang::sym(id_col) might be more appropriate.

Reply:
Changed as suggested.